src/cpp/rtps/xmlparser/XMLEndpointParser.cpp

```diff
@@ -249,6 +249,38 @@ XMLP_ret XMLEndpointParser::loadXMLReaderEndpoint(tinyxml2::XMLElement* xml_endp
                 return XMLP_ret::XML_ERROR;
             }
         }
+        else if (key == TOPIC) // deprecated api
+        {
+            const char *topicName = element->Attribute(NAME);
+            const char *typeName = element->Attribute(DATA_TYPE);
+            const char *kind = element->Attribute(KIND);
+
+            rdata->topicName() = topicName ? std::string(topicName) : std::string("");
```

Comment:
```suggestion
            rdata->topicName() = ((topicName != nullptr) ? std::string(topicName) : std::string(""));
```
Even if this notation is correct, I would use a more explicit one.
Also using the std `nullptr` even when `tinyxml` return 0 instead of `null`.